constrained-decls.html

```diff
@@ -316,6 +316,17 @@ <h2 id="part2">Part 2: Relaxed &ldquo;constrained <code>auto</code>&rdquo;</h2>
   not a replacement for <code>Constraint auto</code>.
 </p>
 
+<p>
+  Trailing return types in function declarations behave the same way:
+</p>
+<blockquote><pre><code>auto f1() -&gt; Constraint; // okay</code></pre></blockquote>
+
+<p>
+  Trailing return types are also used for <i>return-type-requirement</i>s; however, <code>auto</code> does not (currently) go in such a context, whereas constraints <em>do</em>.
```

Comment:
We could explain that we need {foo} -> auto and {foo} -> Constraint here.

Reply:
> We could explain that we mean {foo} -> auto and {foo} -> Constraint here.

I've added examples. The latter case is the status quo, so I've only added the new cases.